src/main/webapp/app/exercises/file-upload/manage/file-upload-exercise-update.component.ts

```diff
@@ -77,10 +84,27 @@ export class FileUploadExerciseUpdateComponent implements OnInit {
         navigateBackFromExerciseUpdate(this.router, this.fileUploadExercise);
     }
 
+    save() {
+        if (this.fileUploadExercise.gradingInstructionFeedbackUsed) {
+            const ref = this.popupService.checkExerciseBeforeUpdate(this.fileUploadExercise, this.backupExercise);
+            if (!this.modalService.hasOpenModals()) {
+                this.saveExercise();
+            } else {
+                ref.then((reference) => {
+                    reference.componentInstance.confirmed.subscribe(() => {
+                        this.saveExercise();
+                    });
+                });
+            }
+        } else {
+            this.saveExercise();
+        }
+    }
```

Comment:
Can this be moved into a service method since the same code is repeated in modeling-exercise-update?

Reply:
this part will changed for second part, I think i can move that part into service.